app/core/Encryptor/Encryptor.test.ts

```
@@ -0,0 +1,243 @@
+      expect(encryptedObject).toHaveProperty('lib', 'original');
+    });
+  });
+
+  describe('decrypt', () => {
+    let decryptAesSpy: jest.SpyInstance,
+      pbkdf2AesSpy: jest.SpyInstance,
+      decryptAesForkedSpy: jest.SpyInstance,
+      pbkdf2AesForkedSpy: jest.SpyInstance;
+
+    beforeEach(() => {
+      decryptAesSpy = jest
+        .spyOn(Aes, 'decrypt')
+        .mockResolvedValue('{"mockData": "mockedPlainText"}');
+      pbkdf2AesSpy = jest
+        .spyOn(Aes, 'pbkdf2')
+        .mockResolvedValue('mockedAesKey');
+      decryptAesForkedSpy = jest
+        .spyOn(AesForked, 'decrypt')
+        .mockResolvedValue('{"mockData": "mockedPlainText"}');
+      pbkdf2AesForkedSpy = jest
+        .spyOn(AesForked, 'pbkdf2')
+        .mockResolvedValue('mockedAesForkedKey');
+    });
+
+    afterEach(() => {
+      decryptAesSpy.mockRestore();
+      pbkdf2AesSpy.mockRestore();
+      decryptAesForkedSpy.mockRestore();
+      pbkdf2AesForkedSpy.mockRestore();
+    });
+
+    it.each([
+      {
+        lib: ENCRYPTION_LIBRARY.original,
+        expectedKey: 'mockedAesKey',
+        expectedPBKDF2Args: ['testPassword', 'mockedSalt', 600000, 256],
+        description:
+          'with original library and default iterations number for key generation',
+        keyMetadata: DEFAULT_DERIVATION_PARAMS,
+      },
+      {
+        lib: ENCRYPTION_LIBRARY.original,
+        expectedKey: 'mockedAesKey',
+        expectedPBKDF2Args: ['testPassword', 'mockedSalt', 5000, 256],
+        description:
+          'with original library and old iterations number for key generation',
+      },
+      {
+        lib: 'random-lib', // Assuming not using "original" should lead to AesForked
+        expectedKey: 'mockedAesForkedKey',
+        expectedPBKDF2Args: ['testPassword', 'mockedSalt'],
+        description:
+          'with library different to "original" and default iterations number for key generation',
+        keyMetadata: DEFAULT_DERIVATION_PARAMS,
```

After doing some refactor to align with `browser-passworder` I found out that this test is "incorrect".

The `AesForked` library actually uses a fixed number of round:
- https://github.com/MetaMask/react-native-aes/blob/master/android/src/main/java/com/tectiv3/aes/RCTAes.java#L34
- https://github.com/MetaMask/react-native-aes/blob/master/ios/RCTAes/lib/AesCrypt.m#L62

Ultimately, we could raise an error if the `keyMetadata` parameters does not match what is being used "underneath".

It has no real impact right now, but just wanted to point that out!